### PR TITLE
Cart events basic documentation

### PR DESCRIPTION
**Pull request by NickWilde1990**

### CODE REVIEW

File: user/pages/03.commerce2/02.developer-guide/04.cart/04.cart-events/docs.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+Drupal Commerce's cart system provides a full range of events for your custom functionality and modules to react to.
+
+The events use Drupal's standard [Symphony based event system](https://www.drupal.org/docs/8/creating-custom-modules/subscribe-to-and-dispatch-events#s-event-systems-overview). So a solid understanding of that is useful. Events are defined in `\Drupal\commerce_cart\Event\CartEvents`.
+
+Events provided:
+- `CartEvents::CART_EMPTY`: Fired after emptying the cart.
+- `CartEvents::CART_ENTITY_ADD`: Fired after adding a purchaseable entity to the cart.
+- `CartEvents::CART_ORDER_ITEM_UPDATE`: Fired after an order item.
+- `CartEvents::CART_ORDER_ITEM_REMOVE`: Fired after removing an order item form the cart.
+- `CartEvents::CART_ITEM_COMPARISON_FIELDS`: Fired when altering the list of comparison fields.
```
Comment (lisastreeter) [inline user/pages/03.commerce2/02.developer-guide/04.cart/04.cart-events/docs.md]: This probably needs a bit more explanation. What are "comparison fields"? Specifically, let's explain that this event can used to alter the "combine items" behavior for matching new items with existing items in the cart.

File: user/pages/03.commerce2/02.developer-guide/04.cart/04.cart-events/docs.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+## Basic Example
+
+This adds a related product when a specified product is added. A reference field - ala commerce_pado is probably better under most situations.
```
Comment (lisastreeter) [inline user/pages/03.commerce2/02.developer-guide/04.cart/04.cart-events/docs.md]: Explanation here is just a little confusing. Maybe change to, "This adds a related product when a specific product is added. In this example, we just hardcode the entity ID of a specific variation. In most situations, you'd probably want to add an entity reference field to your product variations to allow admin users to dynamically specify the "related" product for each variation. For an example, see the [Commerce Product Add-on](https://www.drupal.org/project/commerce_pado) contrib module."

File: user/pages/03.commerce2/02.developer-guide/04.cart/04.cart-events/docs.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+`modulename.services.yml`
+```yaml
+services:
+  modulename.event_subscriber:
+    class: Drupal\modulename\EventSubscriber\EventSubscriber
```
Comment (lisastreeter) [inline user/pages/03.commerce2/02.developer-guide/04.cart/04.cart-events/docs.md]: Let's differentiate the "EventSubscriber" folder from the "EventSubscriber" class name by using a less general name for the example:
Drupal\modulename\EventSubscriber\CartEventSubscriber

File: user/pages/03.commerce2/02.developer-guide/04.cart/04.cart-events/docs.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+use Drupal\commerce_cart\Event\CartEvents;
+use Drupal\commerce_product\Entity\ProductVariation;
+use Drupal\Core\Messenger\MessengerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+
+/**
+ * Cart Event Subscriber.
+ */
+class EventSubscriber implements EventSubscriberInterface {
```
Comment (lisastreeter) [inline user/pages/03.commerce2/02.developer-guide/04.cart/04.cart-events/docs.md]: class CartEventSubscriber implements EventSubscriberInterface {

File: user/pages/03.commerce2/02.developer-guide/04.cart/04.cart-events/docs.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+```yaml
+services:
+  modulename.event_subscriber:
+    class: Drupal\modulename\EventSubscriber\EventSubscriber
+    arguments: ['@messenger', '@commerce_cart.cart_manager']
+    tags:
+      - { name: event_subscriber }
+```
+
+`EventSubscriber.php`
```
Comment (lisastreeter) [inline user/pages/03.commerce2/02.developer-guide/04.cart/04.cart-events/docs.md]: CartEventSubscriber.php

**Comment by NickWilde1990** [comment]
Thanks for the feedback, updated

**Pull request merged**